libs/data/tests/test_utils.py

```diff
@@ -148,9 +150,44 @@ def test_sample_kernels_with_trigger_distance(trigger_distance_size, size):
         # the trigger t0 is half way through timeseries
         t0_value = xsize // 2
 
-        # get the center of the sampled kernel
-        kernel_center_value = (np.max(kernel) - (size // 2)) % xsize
+        # get the distance of the closest sample in the kernel
+        minimum_dist = np.min(np.abs((kernel % xsize) - t0_value))

```

Comment:
I think the logic for implementing both negative, zero, and positive `trigger_distance`s in a single test should be pretty straightforward, something like

```python
diffs = (kernel % xsize) - t0_value
close_edge = np.abs([diffs.min(), diffs.max()]).min()

if trigger_distance_size >= 0:
    assert close_edge <= trigger_distance_size
else:
    assert close_edge >= abs(trigger_distance_size)
```

Then you can just parameterize `trigger_distance_size` either as a fixture or with `@pytest.mark.parametrize("trigger_distance_size", [...])`